### PR TITLE
GMMK Pro: Fixed Bootmagic Lite support; added personal keymap and VIA support

### PR DESCRIPTION
**Pull request by WholesomeDucky**
## Description

Inside pro.h, the keys that are normally "RAlt" and "Fn" were swapped, causing confusion in making keymaps, since Fn needed to be left of RAlt in the keymap when the physical key is to the right.

My personal keymap is also included, which I adjusted alongside my fix for these two keys being swapped.

Default keymap is already correct and does not need to be changed for this (although RGUI is used instead of Fn. Maybe good use of Fn couldn't be made here?)

Added VIA support. JSON made by me is being submitted to VIA repo.

config.h has been changed to have bootmagic row/column defined for this board to use Esc instead of LShift for Bootmagic.

Firmwares in question have been compiled and flashed to my board with my keymap & tested.

First PR, always open to feedback on how to do things better.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by WholesomeDucky** [comment]
I realized in one of my previous commits I said I added 1000hz polling, but forgot to mention in a later commit that I removed this.

**Comment by WholesomeDucky** [comment]
Additional commits added for VIA support. This is tested to the absolute best of my ability and I am submitted a PR for the via repo to have my JSON for the board.

**Comment by drashna** [comment]
> First PR, always open to feedback on how to do things better.

I mean, looks good to me!

**Comment by WholesomeDucky** [comment]
> > First PR, always open to feedback on how to do things better.
> 
> 
> 
> I mean, looks good to me!

Sweet, I'll PR to VIA in about an hour

File: keyboards/gmmk/pro/config.h [hunk 1]
```diff
@@ -44,3 +46,5 @@
 #define LOCKING_SUPPORT_ENABLE
 /* Locking resynchronize hack */
 #define LOCKING_RESYNC_ENABLE
+
+//
```
Comment (noroadsleft) [inline keyboards/gmmk/pro/config.h:50]:
```suggestion
```

File: keyboards/gmmk/pro/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+//      Ct_L     Win_L    Alt_L                               SPACE                               Alt_R    FN       Ct_R     Left     Down     Right
+
+
+    [0] = LAYOUT(
+        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_PSCR,          KC_MUTE,
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,          KC_DEL,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_PGUP,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,           KC_PGDN,
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT, KC_UP,   KC_END,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RGUI, KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
```
Comment (noroadsleft) [inline keyboards/gmmk/pro/keymaps/via/keymap.c:35]: I actually just merged a PR that changed the default keymap to add a `MO(1)` key and a `RESET` keycode.

This suggestion and the next duplicate that change for the VIA keymap.

```suggestion
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, MO(1),   KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
```

File: keyboards/gmmk/pro/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    ),
+	
+    [3] = LAYOUT(
+        KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,  KC_TRNS, KC_TRNS,  KC_TRNS,          KC_TRNS,
+        KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,  KC_TRNS, KC_TRNS,  KC_TRNS,          KC_TRNS,
+        KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,  KC_TRNS, KC_TRNS,  KC_TRNS,          KC_TRNS,
+        KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,  KC_TRNS,           KC_TRNS,          KC_TRNS,
+        KC_TRNS,             KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,  KC_TRNS,           KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS,  KC_TRNS,   KC_TRNS,                                    KC_TRNS,                                    KC_TRNS,  KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS
```
Comment (noroadsleft) [inline keyboards/gmmk/pro/keymaps/via/keymap.c:62]:
```suggestion
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, RESET,            _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,          _______,
        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______,
        _______, _______, _______,                            _______,                            _______, _______, _______, _______, _______, _______
    ),

    [2] = LAYOUT(
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,          _______,
        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______,
        _______, _______, _______,                            _______,                            _______, _______, _______, _______, _______, _______
    ),

    [3] = LAYOUT(
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,          _______,
        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______,
        _______, _______, _______,                            _______,                            _______, _______, _______, _______, _______, _______
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/gmmk/pro/keymaps/via/keymap.c (hunk 3 above)
Comment (WholesomeDucky) [reply keyboards/gmmk/pro/keymaps/via/keymap.c]: All looks good to me!

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**